.github/workflows/release.yml

```diff
@@ -0,0 +1,20 @@
+name: PyPI Release
+
+on:
```

Comment:
My understanding is that it would only publish when we explicitly click on "release" within the Github interface. I would add another trigger: when a tag gets pushed on `main` with a specific tag format (e.g. `v.*.*.*`).

This would enforce the use of tags, which are super helpful when tracking bugs / regressions between versions.

What do you think ?

Reply:
This:

```
on:
  release:
    types: [created]
```

is only triggered when we have a full release (done manually)

Reply:
Tags are enforced by GitHub. When you create a release, it will automatically create a git tag for it, or if you specify an existing git tag, it will assign the release to that tag, so I don't think there is a big difference on what exactly triggers the release. With this trigger, it is possible to push the tag beforehand, try out the tagged build and only make a release if everything works. It's even possible to make Draft releases, which you might not want to publish to PyPI (e.g. alpha, beta releases).